src/components/app/LogsEventEmitter.ts

```diff
@@ -0,0 +1,21 @@
+import { EventEmitter } from '@/originalBudgetTrackingApp';
+import { EventNames } from '@/originalBudgetTrackingApp/eventEmitters/EventEmitter';
+import { Levels, LogEntry } from '../shared/log/types';
+
+export default (callback: (entry: LogEntry) => void) => {
+  const eventPublisher = new EventEmitter.BudgetTrackingEventEmitter();
```

Comment:
I think it would be cleaner to add the level to the BudgetTrackingEvent itself, setting it by default to INFO and having error set it to ERROR. Then you won't need this class and can use the event publisher directly.

In the current implementation you don't have access to the full api of the emitter which is useful (see comment about the option to set the status by listening to events)

Reply:
Also, we are losing data here that doesn't come from the message. For example, it would make sense in the future to display the icon of the account from which this log entry came from,for that you would need the account details but here they are filtered out.

Reply:
You're right, but I think you starting to understand the difficulty. We have so many event types, and if you want to get the account, for example, you have to explicitly say which events are related to account.

I think it is because we are using interfaces, so there is no meaning of the inheritance, actually. With interfaces, you just declare the shape of the object, but you can't know in the listener if the payload is an instance of something.

Reply:
Changing to classes makes sense.

Maybe there are too many event types, but I would spend time on improving that together with a use cases that will use the specific events (like the progress indication)

Reply:
> I think it would be cleaner to add the level to the BudgetTrackingEvent itself, setting it by default to INFO and having error set it to ERROR. Then you won't need this class and can use the event publisher directly.
> 
> In the current implementation you don't have access to the full api of the emitter which is useful (see comment about the option to set the status by listening to events)

https://github.com/brafdlog/budget-tracking/issues/113

Reply:
OK then, let's keep it that way and we will improve it as needed as you suggested